TaylorSource/CoreData/NSFRCDatasource.swift

```diff
@@ -0,0 +1,80 @@
+import UIKit
+import CoreData
+
+public struct NSFRCDatasource<
+    Factory
+    where
+    Factory: _FactoryType,
+    Factory.ViewType: NSFRCIndexedUpdateConsumer,
+    Factory.CellIndexType == NSFRCCellIndex,
+Factory.SupplementaryIndexType == NSFRCSupplementaryIndex>: DatasourceType {
+    
+    public typealias FactoryType = Factory
+    
+    public let identifier: String
+    public let factory: Factory
+    public var title: String? = .None
+    
+    public let updateHandler = NSFRCUpdateHandler()
+    public var selectionManager = IndexPathSelectionManager()
+    
+    public init(id: String, fetchedResultsController: NSFetchedResultsController, factory f: Factory) {
+        fetchedResultsController.delegate = self.updateHandler
+        self.fetchedResultsController = fetchedResultsController
+        identifier = id
+        factory = f
+    }
+    
+    public var numberOfSections: Int {
+        return self.fetchedResultsController.sections?.count ?? 0
```

Comment:
It seems to be the Swift style & preference to only use `self.` inside closures and when necessary due to name clashes. I think it would be good to remove them, it makes for much less visual noise. I'll not mention this for any other occurrences - but there are many. 😁